src/main/java/com/redhat/pantheon/asciidoctor/extension/TemplateDirectory.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import com.redhat.pantheon.sling.PantheonBundleActivator;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Enumeration;
+
+/**
+ * Created by ben on 3/8/19.
+ */
+public class TemplateDirectory {
+
+    private static File tmplDir;
+
+    public static File get() {
+        if (tmplDir == null) {
+            Enumeration<URL> urls = PantheonBundleActivator.getContext().getBundle().findEntries("apps/pantheon/templates/haml/html5", "*", false);
```

Comment:
I feel like we should do this in the Bundle activator itself.

Also, this class looks an awful lot like a singleton, and we should try to avoid singletons in clustered environments.

Reply:
Agree with the bundle activator idea. I will implement that.

Why was the singleton pattern bad in this case though? My logic was that someday this will be deployed on Openshift, and yes the pods won't be able to share this instance, but each pod will need its own set of /tmp template files, so that falls into the "feature not bug" category. Were you thinking of a different angle?

Reply:
You're right; in this case it won't harm us. I am just being cautious that the moment we put mutable shared memory data on a singleton is where we fail. I guess we could store it in the JCR repository itself.